cubedash/summary/_model.py

```diff
@@ -191,3 +206,34 @@ def footprint_srid(self):
 def _has_shape(datasets: Tuple[Dataset, Tuple[BaseGeometry, bool]]) -> bool:
     dataset, (shape, was_valid) = datasets
     return shape is not None
+
+
+# chain all the polygon within Multipolygon into a list
+def _polygon_chain(valid_geometries):
+    polygonlist = []
+    for poly in valid_geometries:
+        if type(poly.footprint_geometry) is MultiPolygon:
+            for p in list(poly.footprint_geometry):
+                polygonlist.append(p)
+        else:
+            polygonlist.append(poly.footprint_geometry)
+    return polygonlist
+
+
+# Recursive filtering for non-noded intersection
+def _filter_geom(geomlist, start=0):
```

Comment:
(same as above.)

Is this docstring correct @pindge, from my quick reading of the code? It's enough non-trivial logic that I think we should give some explanation.

```suggestion
def _filter_geom(geomlist: List[BaseGeometry], start=0) -> List[BaseGeometry]:
    """
    Recursive filtering of un-unionable polygons. Input list is modified in-place.
    
    Exhaustively searches for a run of polygons that cause a union error 
    (eg. "non-noded intersection"), and cuts out the first one that it finds.
    """
```